clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -10908,6 +10908,9 @@ def err_sycl_restrict : Error<
   "|use a const static or global variable that is neither zero-initialized "
   "nor constant-initialized"
   "}0">;
+def err_array_num_elements_exceeded
+    : Error<"Due to SPIR-V intermediate format limitations, constant arrays with number of elements more than 65532 cannot be used in SYCL."
+            "You can workaround this limitation by splitting the array into several ones">;
```

Comment:
I am not too fond of this diagnostic. How about -

       SPIR-V intermediate format limitation (or can we just say spec here?) disallows the use of arrays with number of 
       elements greater than 65532. You can work around this limitation by splitting the array.